protocol/x/clob/e2e/rate_limit_test.go

```diff
@@ -170,6 +195,180 @@ func TestRateLimitingOrders_RateLimitsAreEnforced(t *testing.T) {
 	}
 }
 
+func TestCombinedPlaceCancelBatchCancel_RateLimitsAreEnforced(t *testing.T) {
+	tests := map[string]struct {
+		blockRateLimitConfig clobtypes.BlockRateLimitConfiguration
+		firstBatch           []sdktypes.Msg
+		secondBatch          []sdktypes.Msg
+		thirdBatch           []sdktypes.Msg
+		firstBatchSuccess    []bool
+		secondBatchSuccess   []bool
+		thirdBatchSuccess    []bool
+		lastOrder            sdktypes.Msg
+	}{
+		"Combination Place, Cancel, BatchCancel orders": {
+			blockRateLimitConfig: clobtypes.BlockRateLimitConfiguration{
+				MaxShortTermOrdersAndCancelsPerNBlocks: []clobtypes.MaxPerNBlocksRateLimit{
+					{
+						NumBlocks: 2,
+						Limit:     6, // TODO FIX THIS AFTER SETTLE ON A NUM
+					},
+				},
+			},
+			firstBatch: []sdktypes.Msg{
+				&PlaceOrder_Alice_Num0_Id0_Clob0_Buy5_Price10_GTB20, // 1-weight success @ 1
+				&PlaceOrder_Alice_Num0_Id0_Clob1_Buy5_Price10_GTB20, // 1-weight success @ 2
+				&CancelOrder_Alice_Num0_Id0_Clob0_GTB20,             // 1-weight success @ 3
+			},
+			firstBatchSuccess: []bool{
+				true,
+				true,
+				true,
+			},
+			secondBatch: []sdktypes.Msg{
+				&PlaceOrder_Alice_Num0_Id0_Clob0_Buy5_Price10_GTB23, // 1-weight success @ 4
+				&CancelOrder_Alice_Num1_Id0_Clob0_GTB20,             // 1-weight success @ 5
+				&BatchCancel_Alice_Num0_Clob0_1_2_3_GTB20,           // 2-weight failure @ 7
+				&CancelOrder_Alice_Num0_Id0_Clob0_GTB23,             // 1-weight failure @ 8
```

Comment:
oh interesting - I didn't know we increment the counter for failed txns. I guess this has always be the behavior?